worker/modelworkermanager/shim.go

```diff
@@ -4,19 +4,36 @@ package modelworkermanager
 
 import (
 	"github.com/juju/errors"
+
+	"github.com/juju/juju/controller"
 	"github.com/juju/juju/state"
 )
 
-// StatePoolModelGetter implements ModelGetter in terms of a *state.StatePool.
-type StatePoolModelGetter struct {
+// StatePoolController implements Controller in terms of a *state.StatePool.
+type StatePoolController struct {
 	*state.StatePool
 }
 
-// Model is part of the ModelGetter interface.
-func (g StatePoolModelGetter) Model(modelUUID string) (Model, func(), error) {
+// Model is part of the Controller interface.
+func (g StatePoolController) Model(modelUUID string) (Model, func(), error) {
 	model, ph, err := g.StatePool.GetModel(modelUUID)
 	if err != nil {
 		return nil, nil, errors.Trace(err)
 	}
 	return model, func() { ph.Release() }, nil
 }
+
+// DBLogger returns a database logger for the specified model.
+func (g StatePoolController) DBLogger(modelUUID string) (DBLogger, error) {
+	ps, err := g.StatePool.Get(modelUUID)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	defer ps.Release()
+	return state.NewDbLogger(ps), nil
+}
+
+// Config is part of the Controller interface.
+func (g StatePoolController) Config() (controller.Config, error) {
+	return g.StatePool.SystemState().ControllerConfig()
```

Comment:
SystemState looks like it can return nil.

Reply:
Fair enough. Updated method.